templates/master/01-master-kubelet/_base/units/kubelet.service.yaml

```diff
@@ -40,6 +40,7 @@ contents: |
         --volume-plugin-dir=/etc/kubernetes/kubelet-plugins/volume/exec \
         {{cloudConfigFlag . }} \
         --hostname-override=${KUBELET_NODE_NAME} \
+        --provider-id=${KUBELET_PROVIDERID} \
```

Comment:
Just curious, since you are adding this to all platforms, how does this affect non-AWS platforms? Would this be empty? Does that cause issues?

Reply:
Yes, this would be empty on all platforms aside from AWS.
And judging by the behaviour of the kubelet, this is only assigned to `node.Spec.ProviderID` if is not empty ("").
https://github.com/kubernetes/kubernetes/blob/39c76ba2edeadb84a115cc3fbd9204a2177f1c28/pkg/kubelet/kubelet_node_status.go#L374

So it should be pretty safe.

Reply:
It shouldn't affect the other platforms as the value of `KUBELET_PROVIDERID` will be the empty string https://github.com/kubernetes/kubernetes/blob/39c76ba2edeadb84a115cc3fbd9204a2177f1c28/pkg/kubelet/kubelet_node_status.go#L374